test/kontrol/README.md

```diff
@@ -0,0 +1,200 @@
+The specification is implemented in the form of Foundry tests, however, running these tests with Kontrol ensures that function behaviors are checked for all possible input and storage values (the latter is ensured by the use of `kevm.symbolicStorage(address(dToken))` cheatcode in `setUp()`).
+
+To run these proofs, execute:
+```shell
+kontrol prove --mt GLDTokenTest.test
+```
+
+To make formal reasoning more efficient, these tests are using lemmas provided in the [lemmas.md](../../lemmas.md) file. For more information on lemmas, please refer to [our documentation](https://docs.runtimeverification.com/kontrol/guides/advancing-proofs).
+
+#### 3. [DToken](./proofs/DToken.k.sol) and [ESynth](./proofs/ESynthProof.k.sol)
+
+These proofs exemplify how to verify properties of contracts present in the Euler codebase by adapting the existing Foundry tests to Kontrol.
+
+[ESynthProof.prove_addIgnoredForTotalSupply_onlyOwner](./proofs/ESynthProof.k.sol#L29) helps verify that the `addIgnoredForTotalSupply` function of the `ESynth` contract can only be called by the owner of the contract for all symbolic inputs.
+
+[DTokenProof.prove_Allowance_ReturnsZero](./proofs/DTokenProof.k.sol#L28) verifies that the `allowance` function of the `DToken` contract returns zero for all inputs and storage values (ensured by the use of `kevm.symbolicStorage(address(dToken))` cheatcode in `setUp()`).
+[DTokenProof.prove_Approve_NotSupported](./proofs/DTokenProof.k.sol#L41) verifies that the `approve` function of the `DToken` contract always reverts with a specific (`Errors.E_NotSupported`) custom error.
+
+To run these three proofs in parallel, execute:
+```shell
+kontrol prove --mt prove_Allowance_ReturnsZero --mt prove_Approve_NotSupported --mt prove_addIgnoredForTotalSupply_onlyOwner --workers 3
+```
+
+#### 4. [EVault](./proofs/EVaultProof.k.sol)
+
+`EVaultProof` showcases the external computation feature of Kontrol. Kontrol is both time and resource intensive, but here's a way of saving an arbitrary amount of time when executing a proof in Kontrol.
+This example instructs on how to offload the initial part of your proof computation to Foundry (which is blazing fast) and then incorporate it into a Kontrol proof.
+
+The way we achieve this is via the cheatcodes `vm.stopAndReturnStateDiff` or `vm.dumpState` (undocumented at the time of writing).
+Depending on the cheatcode used the approach is slightly different. However, in both cases we produce a JSON containing the state of the chain after executing some code, and load that state into Kontrol.
+In this example, we are using `vm.dumpState` since it's easier to work with. For complete instructions on how to include external computation with `vm.stopAndReturnStateDiff`, please see the [documentation](https://github.com/runtimeverification/kontrol/tree/master/docs/external-computation).
```

Comment:
I plan to upstream the documentation for `vm.dumpState` into Kontrol either today or tomorrow, so maybe we can delete the mention of undocumentation :slightly_smiling_face:

Reply:
Done, thanks!